tools/generator/Java.Interop.Tools.Generator.Importers/XmlApiImporter.cs

```diff
@@ -13,6 +15,95 @@ class XmlApiImporter
 	{
 		static readonly Regex api_level = new Regex (@"api-(\d+).xml");
 
+		public static List<GenBase> Parse (XDocument doc, CodeGenerationOptions options)
+		{
+			if (doc is null)
+				return null;
+
+			var root = doc.Root;
+
+			if ((root == null) || !root.HasElements) {
+				Report.LogCodedWarning (0, Report.WarningNoPackageElements);
+				return null;
+			}
+
+			var gens = new List<GenBase> ();
+
+			foreach (var elem in root.Elements ()) {
+				switch (elem.Name.LocalName) {
+					case "package":
+						gens.AddRange (ParsePackage (elem, options));
+						break;
+					case "enum":
+						var sym = new EnumSymbol (elem.XGetAttribute ("name"));
```

Comment:
I wonder if we should capture `elem.XGetAttribute("name")` into a `name` variable, so that we don't call it twice.